### ISSUE TITLE
serverless heavy operations (ffmpeg) offloading 

### ISSUE THREAD

**Issue by maratsh** [photoprism/photoprism#1876]
I'm using photoprism  together with other familily members. It's working just fine on scaleway stardust (1CPU/1GB). Hovewer, video and tensorflow related staff are not working due to lack of resources. Ocassionaly, I'm uploading some videos.

 I could upgrade my server to get video feature, but it seems non efficient, as I upload a few videos in a week. 
 My idea is to offload occasional heavy operations to serverless functions (aws lambda or scaleway functions).

So photoprism could take the video, send it to lambda(deployed by me) and use result as it was produced locally.

**Comment by lastzero** [comment]
Performance is also affected by I/O, so using a remote service has its drawbacks too. I don't recommend going for an enterprise software architecture for private use, also because it's hard to debug and security can be easily compromised. 

Suppose someone manipulates the DNS, then your private videos can be sent to the wrong server. Sure, there are ways to avoid this, but in the end it's a lot of work and may not be worth what you save in hosting fees.

**Comment by maratsh** [comment]
Most cloud servers have good connection and I don't mind video being processed being delayed. 
Security issue is common for everything connected to the internet, as server itself could be hacked, network could be hacked etc ...

Debugging in this case is a real issue, I assume. However, multiple open source solutions already exist in this field.

**Comment by Diomas** [comment]
Could it be also an option to use client-side processing with something like ffmpeg+webassembly tensorflow+webasembly?

Let's say I have a pretty performant PC but it's not 100% uptime and PP is hosted on a raspberry. So it would be nice to use my PC for such heavy things while I'm connected to PP.

**Comment by lastzero** [comment]
Yes, that's possible - but [our funding is far too low at the moment](https://docs.photoprism.app/funding/) to experiment with such things. There are a lot of detail problems to solve, which means we need a lot of time.
